src/components/RightDockedModal.js

```diff
@@ -27,23 +32,25 @@ const defaultProps = {
     currentlyViewedReportID: '',
 };
 
-const SettingsModal = props => (
+const RightDockedModal = ({
+    currentlyViewedReportID, isVisible, title, Page,
+}) => (
     <ModalWithHeader
         type={CONST.MODAL.MODAL_TYPE.RIGHT_DOCKED}
-        onClose={() => redirect(_.isEmpty(props.currentlyViewedReportID)
+        onClose={() => redirect(_.isEmpty(currentlyViewedReportID)
             ? ROUTES.HOME
-            : ROUTES.getReportRoute(props.currentlyViewedReportID))}
-        isVisible={props.isVisible}
-        title="Settings"
+            : ROUTES.getReportRoute(currentlyViewedReportID))}
+        isVisible={isVisible}
+        title={title}
         backgroundColor={themeColors.componentBG}
     >
-        <SettingsPage />
+        <Page />
```

Comment:
Since we are rendering this as a child maybe instead of specifying a `Page` prop we can use `this.props.children` allowing this component to render whatever it likes?

Reply:
Done!